pyiron_contrib/atomistic/atomicrex/function_factory.py

```diff
@@ -0,0 +1,476 @@
+        return Spline(identifier, cutoff, derivative_left, derivative_right, species)
+
+    @staticmethod
+    def exp_A_screening(identifier, cutoff, species=["*", "*"], is_screening_function=True):
+        return ExpA(identifier, cutoff, species=species, is_screening_function=is_screening_function)
+    
+    @staticmethod
+    def exp_B_screening(identifier, cutoff, rc, alpha, exponent, species=["*", "*"], is_screening_function=True):
+        return ExpB(identifier, cutoff, rc, alpha, exponent, species=species, is_screening_function=is_screening_function)
+        
+    @staticmethod
+    def exp_gaussian_screening(identifier, cutoff, stddev, alpha, exponent, species=["*", "*"], is_screening_function=True):
+        return ExpGaussian(identifier, cutoff, stddev, alpha, exponent, species=species, is_screening_function=is_screening_function)
+    
+    @staticmethod
+    def morse_A(identifier, D0, r0, beta, species=["*","*"]):
+        return MorseA(identifier, D0, r0, beta, species=species)
+        
+    @staticmethod
+    def morse_B(identifier, D0, r0, beta, S, delta, species=["*","*"]):
+        return MorseB(identifier, D0, r0, beta, S, delta, species=species)
+        
+    @staticmethod
+    def morse_C(identifier, A, B, mu, lambda_val, delta, species=["*", "*"]):
+        return MorseC(identifier, A, B, mu, lambda_val, delta, species=species)
+    
+    @staticmethod
+    def gaussian(identifier, prefactor, eta, mu, species=["*", "*"]):
+        return Gaussian(identifier, prefactor, eta, mu, species)
+
+   
+class SpecialFunction(InputList):
+    def __init__(self, identifier, species=["*", "*"], is_screening_function=False):
+        super().__init__(table_name=f"special_function_{identifier}")
+        self.species = species
+        self.parameters = FunctionParameterList()
+        self.is_screening_function = is_screening_function
+        self.identifier = identifier
+        if not is_screening_function:
+            self.screening = None
+
+    def _to_xml_element(self, name):
+        if self.is_screening_function:
+            screening = ET.Element("screening")
+            root = ET.SubElement(screening, f"{name}")
+        else:
+            root = ET.Element(f"{name}")
+        
+        root.set("id", f"{self.identifier}")
+        for param in self.parameters.values():
+            p = ET.SubElement(root, f"{param.param}")
+            p.text = f"{param.start_val}"
+        root.append(self.parameters.fit_dofs_to_xml_element())
+        if not self.is_screening_function:
+            if self.screening is not None:
+                root.append(self.screening._to_xml_element())
+            return root    
+        else:
+            return screening
+
+
+class Poly(InputList):
+    def __init__(self, identifier, cutoff, species):
+        super().__init__(table_name=f"Poly_{identifier}")
+        self.identifier = identifier
```

Comment:
These will end up as elements in the InputList.  If you don't want this call `object.__setattr__` like [here](https://github.com/pyiron/pyiron_base/blob/a1645298b0d8cb978424749d2ff4cb91d1302ede/pyiron_base/generic/inputlist.py#L155) to define them as instance variables.  Accessing them afterwards works with the normal syntax though.